### PR TITLE
RMW isolation stress test fix : separate cc and py tests added

### PR DESCRIPTION
**Pull request by adityapande-1995**
This PR addresses the comments here : https://github.com/RobotLocomotion/drake-ros/pull/232#issuecomment-1445855433, by adding a py and cc test, each containing talkers, listeners, and process management logic.

The aim is to make the tests more readable, and ensure that the talkers and listeners run in parallel.

### CODE REVIEW

**Review by EricCousineau-TRI — CHANGES_REQUESTED** [review]
First pass done, PTAL!

> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-:-NQ6Ao-t6BEy-O5fICfY:b-v6scke)* status: all files reviewed, 7 unresolved discussions (waiting on @adityapande-1995)

___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 54 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ5zdNw3eereZdvs9zj:-NQ5zdNw3eereZdvs9zk:b-esxpi8) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L54)):*
> ```cpp
>   void topicCallback(const std_msgs::msg::Float64::SharedPtr msg) {
>     if (msg->data != id_) {
>       throw std::runtime_error("I heard '" + std::to_string(msg->data) + "' yet " +
> ```

nit Consider using `fmt::format()` instead of string concatenation + `to_string`.
___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 76 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6-95r9tfIPzrSjyxa:-NQ6-95s5uVgSZzKIxkn:b-8ycg4y) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L76)):*
> ```cpp
> pid_t launch_node(int argc, char* argv[], int id, const std::string& nodeType="talker"){
>   // Launch a new process.
>   auto processId = fork();
> ```

(Here and elsewhere) We should use `snake_case` for (most) variable names.
https://drake.mit.edu/styleguide/cppguide.html#Variable_Names
___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 77 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ5zynIDUohJgmFuc7H:-NQ5zynIDUohJgmFuc7I:ba8650r) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L77)):*
> ```cpp
>   // Launch a new process.
>   auto processId = fork();
>   // Return to the parent process.
> ```

nit Comment a bit hard to parse for me.

Consider clarifying:
```
if (processId != 0) {
  // We are in the parent process. Return the child process id.
  return processId;
}
```
___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 104 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6AbpaBW1kclTE2PGU:-NQ6AbpaBW1kclTE2PGV:b1ut3b) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L104)):*
> ```cpp
> int main(int argc, char* argv[]) {
>   // Number of isolated talker-listener pairs.
>   const int numberOfIsoaltedPairs = 3;
> ```

nit This should be a command-line argument for real stress testing.

Also, should it be consistent w/ the default of `5` below?
___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py` line 51 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6ARdrD4JbjDzRUoXc:-NQ6ARds3ZDCMar7YEc1:b3nb1wk) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py#L51)):*
> ```Python
> def launch_node(id, node_type="talker"):
>     # Create an isolated enviroment.
>     os.environ["PYTHONBUFFERED"] = '0'
> ```

nit Why does this need to be unbuffered?
___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py` line 76 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6AVwWAAkPuMNS1ZXw:-NQ6AVwX5i1ZpRQxSAMv:b74ybaw) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py#L76)):*
> ```Python
>         raise
> 
> if __name__ == "__main__":
> ```

This should live in a `main()` function to avoid accidental scope leakage.
___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py` line 78 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6AZSx6gJZSlfrtUSf:-NQ6AZSy8AdlI8sZ5MRx:b-s6emes) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py#L78)):*
> ```Python
> if __name__ == "__main__":
>     # Number of isolated talker-listener pairs.
>     number_of_isolated_pairs = 5
> ```

nit This should be a command-line argument.

**Comment by adityapande-1995** [comment]
@EricCousineau-TRI I've addressed the comments in the code, but I can't find the comment thread in Reviewable.

**Comment by EricCousineau-TRI** [comment]
> [...] but I can't find the comment thread in Reviewable

Huh, I now look at Reviewable, and I see that they are all draft comments. They may have had a cloud-y blip.

> CI couldn't find `@fmt`, I changed it to linkopts = ["-lfmt"].

Two notes:
- Be wary of `@expr` without escaping. You've now pinged this person: `https://github.com/fmt`
- `bazel_ros2_rules` does not depend on `@drake`, so we must provide it ourselves. That is expected, and "OK" in my book. Will comment on approach.

**Review by EricCousineau-TRI — CHANGES_REQUESTED** [review]
> Reviewed 9 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256)* status: 6 of 9 files reviewed, 7 unresolved discussions (waiting on @adityapande-1995)

**Review by EricCousineau-TRI — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256)* status: 7 of 9 files reviewed, 6 unresolved discussions (waiting on @adityapande-1995)

___
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/package.BUILD.bazel` line 73 at r3](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQAk-3W9j1PnWDo5hgf:-NQAk-3X21QmbUaU2lOX:bg3of6v) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f2a0a7c468330ad7f6fe6be23068569e6e81e0d9/bazel_ros2_rules/ros2/resources/rmw_isolation/package.BUILD.bazel#L73)):*
> ```
>     args = ["5"],
>     rmw_implementation = "rmw_cyclonedds_cpp",
>     linkopts = ["-lfmt"],
> ```

We should ideally use `@fmt`.

Can you add a small repository to the `WORKSPACE` file? e.g.
https://bazel.build/reference/be/workspace#new_local_repository
```
# TODO(eric.cousineau): Replace with proper repository rule.
new_local_repository(
   name = "fmt",
   build_file_contents = """cc_library(name = "fmt", linkopts=["-lfmt"])""",
)
```

If it takes you more than 10min, add a TODO.

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 77 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ5zynIDUohJgmFuc7H:-NQCbZCM83JKrBgK9L-5:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L77)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Context of comment now unclear.

Can you place this comment inside of the `if`?
Alternatively, qualify the comment with `If we are in [...]`.
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 104 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6AbpaBW1kclTE2PGU:-NQCbcLN3qIoybTyG0Lt:b-896fjx) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L104)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

nit This should be a command-line argument for real stress testing.

Also, should it be consistent w/ the default of `5` below?
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 106 at r3](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQAjAkr1Xnvp9G2faO_:-NQCbeRuCGjnVqPoHd78:b-e2pj71) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f2a0a7c468330ad7f6fe6be23068569e6e81e0d9/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L106)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

nit Is it possible to leverage `gflags` here?

If it takes you more than 5min, can you leave a TODO?
</blockquote></details>

Done, added gflags.

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 108 at r3](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQAjahm99d59n57kGui:-NQCbhpX50jEm4PEwUZm:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f2a0a7c468330ad7f6fe6be23068569e6e81e0d9/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L108)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Per below, can you make sure the flag is explicit? i.e. `--number_of_isolated_pairs 5` should be the valid spelling.
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py` line 76 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6AVwWAAkPuMNS1ZXw:-NQCbklmEYjEqY4Lk4yI:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py#L76)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

This should live in a `main()` function to avoid accidental scope leakage.
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py` line 78 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6AZSx6gJZSlfrtUSf:-NQCbnH16fPNi5J27nyX:b-u2oauc) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py#L78)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

nit This should be a command-line argument.
</blockquote></details>

Done using argparse.

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py` line 80 at r3](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQAjLkND5TQhEZ2hxS6:-NQCbrW11KBySv71Rash:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f2a0a7c468330ad7f6fe6be23068569e6e81e0d9/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py#L80)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

nit This should use `argparse`.

Consider making the argument explicit as well, e.g.
```
parser.add_argument("--number_of_isolated_pairs", type=int, default=5)
```
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/package.BUILD.bazel` line 73 at r3](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQAk-3W9j1PnWDo5hgf:-NQCcW5z-jjbf1yAdk4A:b-896fix) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f2a0a7c468330ad7f6fe6be23068569e6e81e0d9/bazel_ros2_rules/ros2/resources/rmw_isolation/package.BUILD.bazel#L73)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

We should ideally use `@fmt`.

Can you add a small repository to the `WORKSPACE` file? e.g.
https://bazel.build/reference/be/workspace#new_local_repository
```
# TODO(eric.cousineau): Replace with proper repository rule.
new_local_repository(
   name = "fmt",
   build_file_contents = """cc_library(name = "fmt", linkopts=["-lfmt"])""",
)
```

If it takes you more than 10min, add a TODO.
</blockquote></details>

Done.

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 54 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ5zdNw3eereZdvs9zj:-NQCcYsL5drD1q98xfRj:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L54)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

nit Consider using `fmt::format()` instead of string concatenation + `to_string`.
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 76 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6-95r9tfIPzrSjyxa:-NQCc_vk23-XU9JQvKo8:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L76)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

(Here and elsewhere) We should use `snake_case` for (most) variable names.
https://drake.mit.edu/styleguide/cppguide.html#Variable_Names
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 108 at r3](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQAjahm99d59n57kGui:-NQCcbXs0nO4XtoZbcgS:b-6onwr) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f2a0a7c468330ad7f6fe6be23068569e6e81e0d9/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L108)):*
<details><summary><i>Previously, adityapande-1995 (Aditya Pande) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done using gflags

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py` line 51 at r1](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQ6ARdrD4JbjDzRUoXc:-NQCcf-ME1Z4NAOtf-Ir:b-ozybyw) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/0b528c16208c7c4e5990ffcf44637d4cdcb47ff4/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.py#L51)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

nit Why does this need to be unbuffered?
</blockquote></details>

Done, removed it.

**Comment by adityapande-1995** [comment]
*[`bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc` line 76 at r3](https://reviewable.io/reviews/RobotLocomotion/drake-ros/256#-NQAixh--zS3jv09UAqE:-NQCcksdEYW-kmMVyLjW:bb6un1j) ([raw file](https://github.com/RobotLocomotion/drake-ros/blob/f2a0a7c468330ad7f6fe6be23068569e6e81e0d9/bazel_ros2_rules/ros2/resources/rmw_isolation/test/rmw_isolation_test.cc#L76)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

nit This should be `CamelCase`:
https://drake.mit.edu/styleguide/cppguide.html#Function_Names
</blockquote></details>

Done

**Comment by adityapande-1995** [comment]
Converting this to a draft untul the rules for fmt and gflags are resolved.

**Comment by adityapande-1995** [comment]
Green CI !
